samplePDF/samplePDFFDBase.cpp

```diff
@@ -1,37 +1,34 @@
 #include "samplePDFFDBase.h"
 
-// Constructors for erec-binned errors
+#include "OscProbCalcer/OscProbCalcerFactory.h"
+#include "Oscillator/OscillatorFactory.h"
+#include "Constants/OscillatorConstants.h"
 
-samplePDFFDBase::samplePDFFDBase(double pot, std::string mc_version, covarianceXsec* xsec_cov)
+#include<algorithm>
+
+
+samplePDFFDBase::samplePDFFDBase(double pot, std::string mc_version_, covarianceXsec* xsec_cov)
   : samplePDFBase()
 //DB Throughout constructor and init, pot is livetime for atmospheric samples
 {
   (void) pot;
-  (void) mc_version;
-  MACH3LOG_INFO("-------------------------------------------------------------------");	
-  MACH3LOG_INFO("Creating samplePDFFDBase object..");
-	
-  //ETA - safety feature so you can't pass a NULL xsec_cov
-  if(xsec_cov == NULL){
-	  MACH3LOG_ERROR("[ERROR:] You've passed me a NULL xsec covariance matrix... I need this to setup splines!");
-	  throw MaCh3Exception(__FILE__ , __LINE__ );}
 
-  samplePDFFD_array = nullptr;
-  samplePDFFD_data = nullptr;
-  oscpars = nullptr;
 
-  //KS: For now FD support only one sample
-  nSamples = 1;
-  SampleName.push_back("FDsample");
-
-  //Default TestStatistic is kPoisson
-  //ETA: this can be configured with samplePDFBase::SetTestStatistic()
-  fTestStatistic = kPoisson;
-
-  //Default values for oscillation-related things
-  doubled_angle = true;
-  osc_binned = false;
-  Osc = NULL;
+  std::cout << "-------------------------------------------------------------------" <<std::endl;
+  MACH3LOG_INFO("Ceating SamplePDFFDBase object");
+    
+  //ETA - safety feature so you can't pass a NULL xsec_cov
+  if(xsec_cov == NULL){
+	MACH3LOG_ERROR("You've passed me a NULL xsec covariance matrix... I need this to setup splines!");
+   	throw MaCh3Exception(__FILE__, __LINE__);
+  }
+  SetXsecCov(xsec_cov);
+  
+  samplePDFFD_array = NULL;
+  samplePDFFD_data = NULL;
+ 
```

Comment:
can we use nullptr, it is genrally recomended for modern c++
https://stackoverflow.com/a/20509811